rcl_sys/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+cmake_minimum_required(VERSION 3.5)
```

Comment:
Given that `rclrs` is deeply tied to `rcl`, I don't think there's a need for a separate `rcl-sys` package. Moreover, while `bindgen` is pretty cool, when I first wrote the Rust bindings I decided to go the `libc` route because that gave me more control over how to expose `rcl` to Rust.

Reply:
rclrs is indeed deeply tied to rcl but I think that the inverse relation is not necessarily true.
Having rcl-sys as a separate crate might allow for different client interface, for example.
Besides, the two crates (rcl-sys and rclrs) work on two well separated level of abstraction and I feel more confortable with keeping the two separate.

I liked how bindgen reduced the volume of Rust code to maintain in the original implementation, plus the lack of C code to write. I feel like it's a whole burden removed from rclrs so I'm interested in more detailed reasons on why not to use it.
What parts of rcl would you like not to expose to Rust and why ?

Reply:
> Having rcl-sys as a separate crate might allow for different client interface, for example.

Given that the only thing that `rcl-sys` does is generating bindings at compile time from `rcl/rcl.h` and `rclutils/error_handling.h`, why not add it as part of `rclrs`? It's not like it's doing much else. I don't think there will be alternative clients based on `rcl-sys`, mostly because it doesn't do much. Moreover, have a look at all the other client libraries: `rclpy`, `rclcpp`, `rcljava`, etc. the code that glues `rcl` and the client APIs are in the wrappers.

> What parts of rcl would you like not to expose to Rust and why?

Allocators, for example. They should be implemented natively in Rust and not exposed via C.

Reply:
> why not add it as part of `rclrs`?

I see your point, I agree that rcl-sys is now empty enough to be merged with rclrs and maybe not that useful on is own. My main motive to split it in the first place was to follow the Rust community guidelines about low level wrapper.

> Allocators, for example. They should be implemented natively in Rust and not exposed via C.

bindgen allows to blacklist specific structs and functions from the generated interface. Do you mean that rcl_get_default_allocator should be blacklisted ? I'm not sure to understand the issue raised by allocators, can you elaborate ?

Reply:
I see pros and cons to both approaches, the biggest issue for bindgen from my point of view is having to install clang for parsing the header files, but it's not such a big deal. In any case, I'm ok with switching to bindgen.